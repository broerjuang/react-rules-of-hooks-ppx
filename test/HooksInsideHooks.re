module GQL = {
  let useMutation =
      (~redirectSuccess=None, ~redirectCancel=None, ~onboarding=false, ()) => {
    let (mutation, _, _) = ApolloHooks.useMutation(~variables, definition);

    () => {
      mutation()
      ->Utils.Promise.map(fst)
      ->Utils.Apollo.Mutation.trackResult(
          ~analyticsMessage=
            onboarding ? "Clicked Pro Plan" : "Clicked Manage Billing",
          ~onData=
            data =>
              switch (
                data##createCheckoutOrBillingPortalSession##checkoutSessionId,
                data##createCheckoutOrBillingPortalSession##billingPortalSessionUrl,
              ) {
              | (Some(checkoutSessionId), None) =>
                Stripe.redirectToCheckout(checkoutSessionId)
              | (None, Some(billingPortalSessionUrl)) =>
                Utils.setLocationUrl(billingPortalSessionUrl)
              | _ =>
                Js.Exn.raiseError(
                  "Exactly one of checkoutSessionId or billingPortalSessionUrl must be defined (not both)",
                )
              },
          (),
        )
      ->ignore;
    };
  };
};
