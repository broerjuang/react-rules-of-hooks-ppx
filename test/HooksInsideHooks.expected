let useMutation =
    (~redirectSuccess=None, ~redirectCancel=None, ~onboarding=false, ()) => {
  let variables =
    makeVariables(
      ~input={
        "redirectSuccess": redirectSuccess,
        "redirectCancel": redirectCancel,
        "onboarding": onboarding,
      },
      (),
    );

  let (mutation, _, _) = ApolloHooks.useMutation(~variables, definition);

  () => {
    mutation()
    ->Utils.Promise.map(fst)
    ->Utils.Apollo.Mutation.trackResult(~analyticsMessage="Clicked")
    ->ignore;
  };
};
